remove code defaulting search bar to repogroup:sample

ijt opened the pull request:
Fixes https://github.com/sourcegraph/sourcegraph/issues/5095

Test plan: manual

attfarhan commented:
I think you'll have to update this PR after Felix's eslint upgrade commit. The code is now in the constructor: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/web/src/search/input/SearchPage.tsx#L41-59.

Also you could remove all code referencing the `repogroup:sample` localStorage key since we won't be using that anymore:
- https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/web/src/search/input/SearchPage.tsx#L146-153


- https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/web/src/search/input/SearchPage.tsx#L39

ijt commented:
> I think you'll have to update this PR after Felix's eslint upgrade commit. The code is now in the constructor: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/web/src/search/input/SearchPage.tsx#L41-59.
> 
> Also you could remove all code referencing the `repogroup:sample` localStorage key since we won't be using that anymore:
> 
> * https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/web/src/search/input/SearchPage.tsx#L146-153
> * https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/web/src/search/input/SearchPage.tsx#L39

Thanks Farhan. I updated it to work with Felix's change and removed those references.

ijt commented:
Checks are passing. PTAL.
